packages/astro/src/content/types-generator.ts

```diff
@@ -419,6 +450,36 @@ async function writeContentFiles({
 				for (const entryKey of Object.keys(collection.entries).sort()) {
 					const dataType = collectionConfig?.schema ? `InferEntrySchema<${collectionKey}>` : 'any';
 					dataTypesStr += `${entryKey}: {\n	id: ${entryKey};\n  collection: ${collectionKey};\n  data: ${dataType}\n};\n`;
+					if (
+						settings.config.experimental.contentCollectionJsonSchema &&
+						collectionConfig?.schema
+					) {
+						let zodSchemaForJson = collectionConfig.schema;
+						if (zodSchemaForJson instanceof z.ZodObject) {
+							zodSchemaForJson = zodSchemaForJson.extend({
+								$schema: z.string().optional(),
+							});
+						}
+						try {
+							await fs.promises.writeFile(
+								new URL(`./${collectionKey.replace(/"/g, '')}.schema.json`, collectionSchemasDir),
+								JSON.stringify(
+									zodToJsonSchema(zodSchemaForJson, {
+										name: collectionKey.replace(/"/g, ''),
+										markdownDescription: true,
+										errorMessages: true,
+									}),
+									null,
+									2
+								)
+							);
+						} catch (err) {
+							logger.warn(
+								'content',
+								`An error was encountered while creating the Json schema. Proceeding without it. Error: ${err}`
```

Comment:
Not sure if this warning also needs rewording. If so please feel free to commit that yourself.

Reply:
JSON is probably all caps when referred to as a name/noun?

Reply:
According to https://json-schema.org/ yes, so I've updated the warning message